SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -1032,6 +1045,55 @@ public Infrastructure.CaseStatusType GetCaseStatusTypeWithFields(string type)
             return response.FirstOrDefault();
         }
 
+        public Infrastructure.CaseStatus CreateCaseStatus(CreateCaseStatusRequest request)
+        {
+
+            var statusType = _databaseContext.CaseStatusTypes
+                .Where(f => f.Name == request.Type)
+                .FirstOrDefault();
+
+            var caseStatus = new Infrastructure.CaseStatus()
+            {
+                PersonId = request.PersonId,
+                TypeId = statusType.Id,
+                StartDate = request.StartDate,
+                Notes = request.Notes,
+                CreatedBy = request.CreatedBy
+            };
+
+            _databaseContext.CaseStatuses.Add(caseStatus);
+            _databaseContext.SaveChanges();
+
+            if (request.Fields != null)
+            {
+                foreach (var optionValue in request.Fields)
+                {
+                    var field = _databaseContext.CaseStatusTypeFields
+                        .FirstOrDefault(f => f.Name == optionValue.Name);
+
+                    var fieldTypeOption = _databaseContext.CaseStatusTypeFieldOptions
+                        .Where(fov => fov.Name == optionValue.Selected)
+                        .FirstOrDefault(fov => fov.TypeFieldId == field.Id);
+
+                    if (fieldTypeOption != null)
+                    {
+                        var fieldOptions = new CaseStatusFieldOption
+                        {
+                            StatusId = caseStatus.Id,
+                            FieldOptionId = fieldTypeOption.Id
+                        };
+
+                        _databaseContext.CaseStatusFieldOptions.Add(fieldOptions);
```

Comment:
instead of adding the field options separately in their own save, you could add the options to the case status object itself like this: `caseStatus.SelectedOptions.Add(fieldOptions);` because the navigation properties are set correctly EF Core is able to create both entities at the same time. This ensures you'll always either get the complete dataset or nothing at all. When saving them separately there's a risk that one save could fail and you would end up with missing data.

Please ensure 'SelectedOptions' is instantiated first if the above approach is used.

Reply:
Oh that's useful, didn't know that!

I will adjust the insert accordingly to this :)